src/include/mpir_handlemem.h

```diff
@@ -231,8 +231,9 @@ static inline void *MPIR_Handle_obj_alloc(MPIR_Object_alloc_t * objmem)
  * Having too large `max_indices` will take more delay to initialize all the objects each
  * time we allocate a new block.
  */
-static inline void *MPIR_Handle_obj_alloc_unsafe(MPIR_Object_alloc_t * objmem,
-                                                 int max_blocks, int max_indices)
+static inline void *MPIR_Handle_obj_alloc_unsafe_impl(MPIR_Object_alloc_t * objmem,
+                                                      int max_blocks, int max_indices,
+                                                      bool skip_finalize)
```

Comment:
Instead of change the function name, how about just add the extra parameter and modify the call-site? I only see two call sites anyway.

And just add a separate `MPIR_Info_create` function that calls `MPIR_Handle_obj_alloc_unsafe(..., true)`.

Reply:
Thanks. That's a good idea! I will change so.

Reply:
I fixed it. No `MPIR_Handle_obj_alloc_unsafe_impl()`.

`MPIR_Handle_obj_alloc_unsafe()` checks `obj_kind` instead of taking another argument since this check does not in the critical path. `MPIR_Handle_obj_free_unsafe()` needs specialization for `MPI_Info` and it is in the critical path, so I add another argument for `MPIR_Handle_obj_free_unsafe()`, which will be inlined and thus the branch will be removed if the handle is not `MPI_Info`.